dags/app/user_profile/udf.py

```diff
@@ -0,0 +1,327 @@
+        schema=[
+            bigquery.SchemaField(task_type, "STRING"),
+            bigquery.SchemaField("category", "INT64"),
+        ],
+    )
+    job = client.load_table_from_dataframe(df, table_id, job_config=job_config)
+    job.result()
+
+
+def chunk_data(data: list[str], batch_size: int) -> Generator[list[str], None, None]:
+    it = iter(data)
+    while batch := list(islice(it, batch_size)):
+        yield batch
+
+
+def get_gemini_response(
+    model: str, max_output_tokens_on_model: int, prompt: str
+) -> str | None:
+    client = genai.Client(api_key=get_gemini_api_key())
+    response = client.models.generate_content(
+        model=model,
+        contents=[prompt],
+        config=types.GenerateContentConfig(
+            max_output_tokens=max_output_tokens_on_model
+        ),
+    )
+    return response.text
+
+
+def process_table(
+    model: str, max_output_tokens_on_model: int, batch_size: int, task_type: str
+) -> None:
+    # Read the input file
+    data_list = read_kktix_ticket_user_profile(task_type)
+    results_to_update = []
+
+    logging.info(
+        f"共讀取到 {len(data_list)} 筆記錄，將分成約 {len(data_list) // batch_size + (1 if len(data_list) % batch_size > 0 else 0)} 個批次處理..."
+    )
+
+    # Process in batches
+    for index, chunk in enumerate(chunk_data(data_list, batch_size), start=1):
+        batch_text = json.dumps(chunk, ensure_ascii=False)
+        batch_prompt = f"{get_task_config(task_type)}\n資料列表:\n{batch_text}"
+
+        logging.info(f"正在處理批次 {index}...")
+
+        # Check batch size against model limit (optional, but good practice)
+        # Note: Token count is not a simple character count, this is a rough check
+        if len(batch_text) > max_output_tokens_on_model * 0.8:  # Use a buffer
+            logging.warning(
+                f"警告：批次 {index} 的大小可能接近或超過模型的最大輸入限制 ({max_output_tokens_on_model} tokens)。"
+            )
+
+        try:
+            response = get_gemini_response(
+                model, max_output_tokens_on_model, batch_prompt
+            )
+            if response:
+                try:
+                    clean_text = response.strip()
+                    clean_text = clean_text[len("```json") :].strip()
+                    clean_text = clean_text[:-3].strip()
+                    result_json = json.loads(clean_text)
+                except json.JSONDecodeError:
+                    raise ValueError(f"批次 {index} JSON解析錯誤：{clean_text}")
+            else:
+                result_json = []
+            results_to_update.extend(result_json)
+
+        except (GoogleAPICallError, DeadlineExceeded, ServiceUnavailable) as e:
+            logging.error(f"批次 {index} API 呼叫錯誤: {e}")
+        except requests.exceptions.RequestException as e:
+            logging.error(f"批次 {index} 網路請求錯誤: {e}")
+        except ValueError as e:
+            logging.error(f"批次 {index} 資料格式錯誤: {e}")
+        except Exception:
+            logging.exception(f"批次 {index} 未知錯誤")
```

Comment:
```suggestion
        except (GoogleAPICallError, DeadlineExceeded, ServiceUnavailable):
            logging.exception(f"批次 {index} API 呼叫錯誤: ")
        except requests.exceptions.RequestException:
            logging.exception(f"批次 {index} 網路請求錯誤: ")
        except ValueError:
            logging.exception(f"批次 {index} 資料格式錯誤: ")
        except Exception:
            logging.exception(f"批次 {index} 未知錯誤")
```

log the whole traceback might make it easier to debug